app/models/ems_azure.rb

```diff
@@ -29,17 +30,16 @@ def connect(options = {})
     self.class.raw_connect(clientid, clientkey, tenant_id)
   end
 
-  def verify_credentials(_auth_type = nil, _options = {})
-    # TODO
-    true
-  end
-
-  def tenant_id=(tenant_id)
-    self.uid_ems = tenant_id
-    save
-  end
+  def verify_credentials(*)
+    begin
```

Comment:
There's a dangling `true` down there after the begin/rescue block.  Move that directly after the call to `connect` and then do what @Fryguy said.